core-graphics/src/image.rs

```diff
@@ -121,6 +122,17 @@ impl CGImageRef {
         };
         data_provider.copy_data()
     }
+
+    /// Returns a cropped image. If the `rect` specifies a rectangle which lies outside of the
+    /// image bounds, the `None` is returned.
+    pub fn cropped(&self, rect: CGRect) -> Option<CGImage> {
```

Comment:
It might be good to include the fact that this uses CGImageCreateWithImageInRect and I'm pretty sure that makes it so that the underlying storage is shared between the two images.